app/views/NewServerView/ServerInput/index.tsx

```diff
@@ -72,14 +74,15 @@ const ServerInput = ({
 						{ backgroundColor: themes[theme].backgroundColor, borderColor: themes[theme].separatorColor }
 					]}
 				>
-					<FlatList
-						data={serversHistory}
-						renderItem={({ item }) => (
+					{/* NB: in theory the .slice(0, MAX_ITEMS_HISTORY) is useless because the number of items in the history is already limited to three when querying the database: */}
+					{/*  https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/5aaf7af5f4004e8b0b9386da579ae4c8aed3a9a0/app/views/NewServerView/index.tsx#L160 */}
+					{/* But we repeat the filtering here, since having more than three items would break the display pretty badly. */}
```

Comment:
You have a point.
That's a weird colocation imo.
What if you write a custom hook to fetch servers history?
You could remove it from NewServerView/index and place it here only.
It only matters here really.

The max items logic would be only inside the custom hook.
wdyt?

Reply:
@diegolmello true. So you would move all the logic around the servers history down in the `<ServerInput />` component?

Reply:
@diegolmello I'll give it a try on this PR. I'll add a separate commit.

Reply:
Hi @diegolmello,

I moved the logic to the <ServerInput /> component, as discussed. The PR is ready for a new review.